### PR TITLE
Cross group transaction

### LINKED ISSUE #183: 🚀 [Feature] Cross Community User Identification
## :rocket: Feature

How do we identify user across communities? The technical answer is the public key, but how do we transfer money, if I have a community link on an user?

Consider:

1. UserA gives UserB his link : mycommunity.a/u/usera
2. UserB is fro another community: ourcommunity.b/u/userb
3. How can UserB Send GDD to UserA having his link only

We need a protocol how to transfere the public key

### PR DESCRIPTION
**Pull request by einhornimmond**
## 🍰 Pullrequest
Stuff which will be needed for cross group transactions
- get group alias from community server api
- get public key for username and group_alias without session from login server api

### Issues
- fixes #183

### Todo
- [ ] Tests

### CODE REVIEW

File: community_server/src/Controller/AppRequestsController.php [hunk 1]
```diff
@@ -156,9 +169,6 @@ public function sendCoins()
         if($required_fields !== true) {
             return $this->returnJson($required_fields);
         }
-        if(!isset($params['memo']) || strlen($params['memo']) < 5 || strlen($params['memo']) > 150) {
-            return $this->returnJson(['state' => 'error', 'msg' => 'memo is not set or not in expected range [5;150]']);
-        }
```
Comment (ulfgebhardt) [inline community_server/src/Controller/AppRequestsController.php:161]: What happend to those restrictions?

File: docu/login_server.api.md [hunk 2]
```diff
@@ -268,6 +271,7 @@ In case of success:
 		"pubkeyhex": "131c7f68dd94b2be4c913400ff7ff4cdc03ac2bda99c2d29edcacb3b065c67e6",
 		"first_name": "Max",
 		"last_name": "Musterman",
+		"email": "max.musterman@gmail.de"
```
Comment (ulfgebhardt) [inline docu/login_server.api.md:274]:
```suggestion
		"email": "max.musterman@gmail.de",
```

**Review by ulfgebhardt — CHANGES_REQUESTED** [review]
This seems to be valid changes, but I fail to understand how it solves #183 since we can still not identify a user across communities with one identifier. Furthermore it would be nice to have this in a human readable form.

How do I identify User B from Community C? Does he have a cross-community-ident-hash? does he have a globally unique username? Do we like combine username & community id into one string? How about anonymity?

**Comment by einhornimmond** [comment]
For #183:
With #484 to make Cross Group Transactions frontend can:
call: GET mycommunity.a/api/getGroupAlias 
to get Group alias from target community, for example "gdd1"
then call
POST mycommunity.a/login-api/getUserInfos with 
```json 
{
	"username": "usera",
	"group_alias": "gdd1",
	"ask": [
		"user.pubkeyhex",
  	]
}
```
to get public key from usera in hex format. 
And now it has all needed data for call to sendCoins

File: community_server/src/Controller/AppRequestsController.php (hunk 1 above)
Comment (einhornimmond) [reply community_server/src/Controller/AppRequestsController.php:161]: The public key is the one identifier which works across communities.
And I have added functions to get the public key from users from another community if you know the community server url and the username, like in your example.

Comment (einhornimmond) [reply community_server/src/Controller/AppRequestsController.php:161]: But in my plan Login-Server is able to get for the username the correct public key from blockchain of the target community. 
We will see if that is also possible with using Iota.

Comment (einhornimmond) [reply community_server/src/Controller/AppRequestsController.php:161]: Oh, you are right I have make mistake there. 
They moved to parseParameterForCreateTransaction
but I have forget the strlen destrictions.

**Pull request closed without merging**